File: app/src/main/java/com/nytimes/android/external/register/di/SchedulerProvider.java
package com.nytimes.android.external.register.di;

import io.reactivex.Scheduler;

/**
 * Helper interface to provide {@link Scheduler} for Rx calls. Extracting scheduler in a interace
 * allows us to inject {@link Scheduler}s for testing.
 */
public interface SchedulerProvider {
    Scheduler getMainThread();
    Scheduler getComputationThread();
    Scheduler getIoThread();
}

File: app/src/main/java/com/nytimes/android/external/register/di/SchedulerProvider.kt
package com.nytimes.android.external.register.di

import io.reactivex.Scheduler

/**
 * Helper interface to provide [Scheduler] for Rx calls. Extracting scheduler in a interace
 * allows us to inject [Scheduler]s for testing.
 */
interface SchedulerProvider {

    fun getMainThread() : Scheduler

    fun getComputationThread() : Scheduler

    fun getIoThread() : Scheduler

}
